### PR TITLE
Put define-class tests into a single suite file

### PR DESCRIPTION
**Pull request by jbouwman**
Combine the programs and error messages in tests/parser-test-files/good-files/define-class.coal and tests/parser-test-files/bad-files/define-class.* into define-class.txt

A structured input format is documented in tests/parser-test-files/README.md. It supports:

- reexecution of single cases
- selective test disabling
- optional error message checking
- test numbering

And provides a place to put additional test metadata.
The tests now compile their inputs, in addition to parsing.

This incorporates changes from #1272 but only converts define-class tests and leaves others intact.

### CODE REVIEW

File: tests/parser-test-files/define-class.txt [hunk 1]
```diff
@@ -13,3 +336,154 @@ error: Malformed class definition
...
+(package test-package)
+
+(define-class (C :a :b (-> :b :c)))
+
+--------------------------------------------------------------------------------
+
+error: Malformed functional dependency
+  --> test:3:23
+   |
+ 3 |  (define-class (C :a :b (-> :b :c)))
```
Comment (Izaakwltn) [inline tests/parser-test-files/define-class.txt:436]: This too, it should probably refer to the arrow, similarly to the => errors above .

(Once again, probably a separate PR)

File: tests/parser-test-files/define-class.txt [hunk 2]
```diff
@@ -1,5 +1,328 @@
...
+
+(define-class (C :a :b =>))
+
+--------------------------------------------------------------------------------
+
+error: Malformed class definition
+  --> test:3:14
+   |
+ 3 |  (define-class (C :a :b =>))
+   |                ^^^^^^^^^^^^ missing class name
```
Comment (Izaakwltn) [inline tests/parser-test-files/define-class.txt:233]: (subsequent PR) this is a suboptimal error, the class name is right there

File: tests/parser-test-files/define-class.txt [hunk 3]
```diff
@@ -13,3 +336,154 @@ error: Malformed class definition
...
+
+(define-class (C :a :b (:a -> :b) :c))
+
+--------------------------------------------------------------------------------
+
+error: Malformed functional dependency
+  --> test:3:34
+   |
+ 3 |  (define-class (C :a :b (:a -> :b) :c))
+   |                                    ^^ expected a list
```
Comment (Izaakwltn) [inline tests/parser-test-files/define-class.txt:389]: One more- I'm not sure what this error *should* be but this seems not ideal

File: tests/parser-test-files/define-class.txt [hunk 4]
```diff
@@ -13,3 +336,154 @@ error: Malformed class definition
...
+
+(define-class (Eq :a => (C :a) :b))
+
+--------------------------------------------------------------------------------
+
+error: Malformed class definition
+  --> test:3:31
+   |
+ 3 |  (define-class (Eq :a => (C :a) :b))
+   |                                 ^^ unexpected form
```
Comment (Izaakwltn) [inline tests/parser-test-files/define-class.txt:472]: Would be better to have a "trailing" error, though might be awkward to implement

**Review by Izaakwltn — APPROVED** [review]
Looks good!

I left a number of comments about error messages in the bad tests- these are probably out of the scope of this PR, but I thought it was a convenient place to point them out.

File: tests/loader.lisp [hunk 5]
```diff
@@ -91,15 +70,26 @@
                            (subseq string 0 end)
                            string))))))
 
+(defun read-flags (string)
+  "Read the first parenthesis-delimited form appearing in STRING, or nil."
+  (let ((open (position-if (lambda (c) (char= c #\()) string))
```
Comment (stylewarning) [inline tests/loader.lisp:75]: `position #\( ...`

**Review by stylewarning — APPROVED** [review]
a small nit on POSITION-IF

File: tests/loader.lisp (hunk 5 above)
Comment (jbouwman) [reply tests/loader.lisp]: fixed, thanks!

**Pull request merged**